src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNamesTests.cs

```diff
@@ -0,0 +1,387 @@
+        Me.firstField = firstField
+        Me.secondField = secondField
+    End Sub
+End Class");
+        }
+
+        private static async Task VerifyCSharpAnalyzerAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithSystemTextJson50,
+                TestCode = source,
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(expected);
+
+            await csharpTest.RunAsync();
+        }
+
+        private static async Task VerifyCSharp9AnalyzerAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp9,
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithSystemTextJson50,
+                TestCode = source,
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(expected);
+
+            await csharpTest.RunAsync();
+        }
+
+        private static async Task VerifyBasicAnalyzerAsync(string source, params DiagnosticResult[] expected)
+        {
+            var basicTest = new VerifyVB.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithSystemTextJson50,
+                TestCode = source,
+            };
+
+            basicTest.ExpectedDiagnostics.AddRange(expected);
+
+            await basicTest.RunAsync();
+        }
+
+        private DiagnosticResult CA1071CSharpPropertyResultAt(int line, int column, params string[] arguments)
+#pragma warning disable RS0030 // Do not used banned APIs
+           => VerifyCS.Diagnostic(ConstructorParametersShouldMatchPropertyNamesAnalyzer.PropertyRule)
+               .WithLocation(line, column)
+#pragma warning restore RS0030 // Do not used banned APIs
+               .WithArguments(arguments);
+
+        private DiagnosticResult CA2243BasicPropertyResultAt(int line, int column, params string[] arguments)
+#pragma warning disable RS0030 // Do not used banned APIs
+            => VerifyVB.Diagnostic(ConstructorParametersShouldMatchPropertyNamesAnalyzer.PropertyRule)
+                .WithLocation(line, column)
+#pragma warning restore RS0030 // Do not used banned APIs
+                .WithArguments(arguments);
+        private DiagnosticResult CA1071CSharpFieldResultAt(int line, int column, params string[] arguments)
+#pragma warning disable RS0030 // Do not used banned APIs
+           => VerifyCS.Diagnostic(ConstructorParametersShouldMatchPropertyNamesAnalyzer.FieldRule)
+               .WithLocation(line, column)
+#pragma warning restore RS0030 // Do not used banned APIs
+               .WithArguments(arguments);
+
+        private DiagnosticResult CA2243BasicFieldResultAt(int line, int column, params string[] arguments)
+#pragma warning disable RS0030 // Do not used banned APIs
```

Comment:
New tests shouldn't use banned APIs.

Reply:
I've quickly tried to use `WithLocation(markupKey)` (not banned API) here instead but without luck yet - tests failed with `System.InvalidOperationException : The markup location '#0' was not found in the input.`. And I'm not sure how to fix it yet. It may be related to [#624 in RoslynSDK](https://github.com/dotnet/roslyn-sdk/issues/624) issue. I'll come back to it later after adding a code fix.

Reply:
@psxvoid Did you add the span where the diagnostic is expected between `{|#0:` and `|}`

Reply:
Yep, you are right, I was missing those. Thank you, fixed.